distribution/sender/routes/debezium-route.xml

```diff
@@ -39,52 +47,60 @@
                         </otherwise>
                     </choice>
                 </when>
-
-                <log message="${exchangeProperty.event}" loggingLevel="DEBUG" />
-
-                <setProperty name="route-retry-count-map">
-                    <spel>#{new java.util.HashMap()}</spel>
-                </setProperty>
-                <split>
-                    <simple>{{db-event.destinations}}</simple>
-                    <script>
-                        <spel>#{getProperty('route-retry-count-map').put(body.trim(), 0)}</spel>
-                    </script>
-                </split>
-                
-                <setProperty name="table-names">
-                    <!-- We need to look up failed events for rows in both the parent and subclass tables -->
-                    <method beanType="org.openmrs.eip.component.utils.Utils" method="getTablesInHierarchy(${exchangeProperty.event.tableName})" />
-                </setProperty>
-
-                <toD uri="jpa:SenderRetryQueueItem?query=SELECT r from SenderRetryQueueItem r WHERE r.event.tableName IN (${exchangeProperty.table-names}) AND r.event.primaryKeyId='${exchangeProperty.event.primaryKeyId}'" />
-
-                <split stopOnException="true">
-                    <simple>${body}</simple>
-                    <choice>
-                        <when>
-                            <simple>${exchangeProperty.route-retry-count-map.containsKey(${body.route.trim()})} != true</simple>
-                            <throwException exceptionType="java.lang.Exception" message="No listener route found with name ${body.route.trim()}" />
-                        </when>
-                        <otherwise>
-                            <script>
-                                <spel>#{getProperty('route-retry-count-map').put(body.route, getProperty('route-retry-count-map').get(body.route.trim()) + 1)}</spel>
-                            </script>
-                        </otherwise>
-                    </choice>
-                </split>
+                <otherwise>
```

Comment:
You will need to remove all this, this route should only construct the DebeziumEvent object and save it to the database

Reply:
The remain logic has to move to the debezium-event-reader route

Reply:
I see... But my problem here is how to reconstruct the record data from DebezimEvent.stateAfter or DebezimEvent.stateBefore. An clue?

Reply:
If you save the states to the DB, you should be able to read them from the DB

Reply:
But what if the operation is DELETE?

Reply:
If we go the route of extracting and saving the uuid like we discussed on the call, then we don't need to store the states, replaces the columns with a single uuid column since it's the only thing we currently look up from the before/after states